### PR TITLE
fix(lib-user): use next/link for internal app routes

### LINKED ISSUE #6350: Full page refresh when following links in the new root app
## Package
_Examples_
- app-root

## Describe the bug
Next.js builds and deploys `app-root` as a single page React app, which should use client-side navigation (`next/link`) to move between pages/views. When I click 'More stats' to go to the user stats page, or 'Back' to go to the home page, the browser performs a full-page refresh, which downloads and mounts the entire React app, even though it's already running.

The React app that Next.js builds for `app-root` is ~2MB of JS, so there's a lot of redundant requests for JS files when you load a new page from scratch. Also some extra delay while the React app mounts and hydrates on each page.

## To Reproduce
1. Go to the Zooniverse home page.
2. Click on More Stats.
3. See that the browser performs a full page refresh.
4. Click Back.
5. See that the browser performs a full page refresh again.
6. Go back to your user stats page and click Generate Volunteer Certificate.
7. That link also performs a full-page refresh and re-mounts the React app.

## Expected behavior
Either use client-side routing between pages/views in your Next.js Single Page App, or (if possible) configure the build so that it doesn't build an SPA.

## Additional context
To see how navigation should work in a Next.js app, go to https://www.zooniverse.org/about and click on the various links to pages within the About section. Each of those links uses client-side routing to update the page, rather than linking to a new page. The React app loads and mounts once, when you enter the About section, then uses client-side routing thereafter.

### PR DESCRIPTION
**Pull request by eatyourgreens**
Use `next/link` for the home page link to user stats, the link to your stats certificate, and the Back link at the top of each page.

Also fix the `lib-user` peer dependencies, which were still requiring Next.js 13.5.

_Please request review from `@zooniverse/frontend` team or an individual member of that team._ 

## Package
- lib-user

## Linked Issue and/or Talk Post
- fixes #6350.

## How to Review
'More stats', on the home page, and 'View certificate', on the user stats page, and the Back link on each of those pages, should now use client-side routing in the root Next.js app.

Build the user library with these changes, then build and run the root app with `yarn build && yarn start`, to verify that those links are now using client-side routing (ie. the page doesn't reload, and `app-root` doesn't unmount and remount in the browser, when moving between views.)

This should also lower the load on the Next.js app server, since it doesn't have to serve the whole app on every page transition. Locally, you'll probably see that as faster page transitions with a production build. 

NB. Next.js development mode disables client-side routing, so I don't think this PR can be checked with `yarn dev`.

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General
- [ ] Tests are passing locally and on Github
- [ ] Documentation is up to date and changelog has been updated if appropriate
- [ ] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected
- [ ] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)
- [ ] FEM works in a mobile browser

## General UX
Example Staging Project: [i-fancy-cats](https://local.zooniverse.org:3000/projects/brooke/i-fancy-cats)
- [ ] All pages of a FEM project load: Home Page, Classify Page, and About Pages
- [ ] Can submit a classification
- [ ] Can sign-in and sign-out
- [ ] The component is accessible
  - Can be used with a screen reader [BBC guide to VoiceOver](https://bbc.github.io/accessibility-news-and-you/assistive-technology/testing-steps/voiceover-mac.html)
  - Can be used from the keyboard [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - It is passing accessibility checks in the storybook

## Bug Fix
- [ ] The PR creator has listed user actions to use when testing if bug is fixed
- [ ] The bug is fixed
- [ ] Unit tests are added or updated

### CODE REVIEW

File: packages/lib-user/src/components/shared/MainContent/MainContent.js [hunk 1]
```diff
@@ -291,7 +292,7 @@ function MainContent({
             margin={{ top: 'small' }}
           >
             <StyledCertificateButton
-              forwardedAs='a'
+              forwardedAs={Link}
```
Comment (eatyourgreens) [inline packages/lib-user/src/components/shared/MainContent/MainContent.js:295]: Noting here that `forwardedAs='a'` does nothing on a Grommet Button. Grommet automatically renders Button as a link, if it has a `href` attribute.

**Comment by goplayoutside3** [comment]
I agree with these changes, but just a heads up that full review won't take place until next week. There are changes to AFD cache rules and actions to purge AFD going into place during FEM deploy tomorrow, so I just want to isolate the changes to prefetching in this PR for deploy the next time instead.

I included the bump in lib-user's peer dependency for Next.js in https://github.com/zooniverse/front-end-monorepo/pull/6367.

**Comment by eatyourgreens** [comment]
Sounds good. Here's a quick screen recording to show the instantaneous transitions between page views on this branch, once the initial data load is complete.


https://github.com/user-attachments/assets/f38dd093-2c80-489f-b941-a38d45eec4be

**Review by goplayoutside3 — APPROVED** [review]
Thanks! I added `next/link` to the See All link to /groups on the homepage too.

**Pull request merged**